### PR TITLE
fix: inline all operation types + handle edge cases where generated t…

### PR DESCRIPTION
**Pull request by n1ru4l**
Closes https://github.com/dotansimha/graphql-code-generator/issues/6149

This pr adds the option `inlineFragmentTypes` which is set to `true` per default.

When `inlineFragmentTypes` is set to `true` the generated operation types will deep inline the fragment selection set types instead of combining the fragment type.

### CODE REVIEW

File: packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts [hunk 1]
```diff
@@ -470,28 +525,30 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
...
+          if (relevant.length === 0) {
+            return null;
+          } else if (relevant.length === 1) {
+            return relevant[0];
+          } else {
+            return `( ${relevant.join(' & ')} )`;
+          }
+        })
+        .filter(Boolean)
+        .join(' | ') + this.getEmptyObjectTypeString(mustAddEmptyObject)
```
Comment (n1ru4l) [inline packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts:542]: Need to add a test to check whether this is correctly omitted if __typename is selected.

File: .changeset/many-socks-warn.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+'@graphql-codegen/flow-operations': patch
+'@graphql-codegen/visitor-plugin-common': patch
+'@graphql-codegen/typescript-operations': patch
+---
+
+Add option `inlineFragmentTypes` for deep inlining fragment types within operation types. This `inlineFragmentTypes` is set to `true` by default.
```
Comment (n1ru4l) [inline .changeset/many-socks-warn.md:7]: @dotansimha Thoughts on the option? I think if we would instead make it a string: `fragmentSpreadTypeMode` with values (`inline`, `fragment`), we can later add `mask` without introducing a new conflicting option

File: dev-test/githunt/__generated__/comment.query.stencil-component.tsx [hunk 3]
```diff
@@ -15,7 +15,13 @@ declare global {
     entry?: Types.Maybe<
       { __typename?: 'Entry' } & Pick<Types.Entry, 'id' | 'createdAt' | 'commentCount'> & {
           postedBy: { __typename?: 'User' } & Pick<Types.User, 'login' | 'html_url'>;
-          comments: Array<Types.Maybe<{ __typename?: 'Comment' } & CommentsPageCommentFragment>>;
+          comments: Array<
+            Types.Maybe<
```
Comment (dotansimha) [inline dev-test/githunt/__generated__/comment.query.stencil-component.tsx:19]: So we still have the fragment `CommentsPageCommentFragment` available as type, and it matches to the type here, right?

File: .changeset/many-socks-warn.md (hunk 2 above)
Comment (dotansimha) [reply .changeset/many-socks-warn.md:7]: I think `inlineFragmentTypes` or `inlineFragments` should be fine

**Review by dotansimha — APPROVED** [review]
Is there any concern for a breaking change here? I know the types are more accurate, but we kept the fragment type the same, so it's still matches on places where we inlined it?

File: dev-test/githunt/__generated__/comment.query.stencil-component.tsx (hunk 3 above)
Comment (n1ru4l) [reply dev-test/githunt/__generated__/comment.query.stencil-component.tsx:19]: yes! the fragment types are still generated, and the types match!

**Pull request merged**